### PR TITLE
feat: Configurable polaris schema

### PR DESCRIPTION
**Pull request by ronkatz96**

### CODE REVIEW

File: packages/polaris-schema/src/scalars/polaris-scalars-resolvers.ts [hunk 1]
```diff
@@ -33,9 +33,16 @@ import {
 } from 'graphql-scalars';
 import { IResolvers } from 'graphql-tools';
 
-export const scalarsResolvers: IResolvers = {
+export const getScalarsResolvers = (shouldAddGraphQLScalars: boolean): IResolvers => {
+    return shouldAddGraphQLScalars ? polarisScalarsResolvers : defaultPolarisScalarsResolvers;
+};
+
+export const defaultPolarisScalarsResolvers: IResolvers = {
```
Comment (chenshoo) [inline packages/polaris-schema/src/scalars/polaris-scalars-resolvers.ts:16]: where is upload

File: packages/polaris-schema/src/utils/merge-types.ts [hunk 2]
```diff
@@ -22,3 +29,17 @@ export const getMergedPolarisTypes = (types: ITypeDefinitions): string =>
             all: true,
         },
     );
+};
+
+function getDirectivesAndScalarsTypeDefs(polarisSchemaConfig: PolarisSchemaConfig) {
+    const directivesAndScalarsTypeDefs: any[] = [];
+    if (polarisSchemaConfig.addPolarisDirectives) {
+        directivesAndScalarsTypeDefs.push(directivesTypeDefs);
```
Comment (chenshoo) [inline packages/polaris-schema/src/utils/merge-types.ts]: in my opinion you should name `directivesTypeDefs` to permissionTypeDefs and add it only if the permissionconfig was provided

File: packages/polaris-schema/src/utils/executable-schema-creator.ts [hunk 3]
```diff
@@ -10,15 +10,20 @@ import {
...
     schemaDirectives?: { [name: string]: typeof SchemaDirectiveVisitor },
 ): GraphQLSchema {
-    const mergedTypes = getMergedPolarisTypes(typeDefs);
-    const mergedResolvers = getMergedPolarisResolvers(resolvers);
+    const mergedTypes = getMergedPolarisTypes(polarisSchemaConfig, typeDefs);
+    const mergedResolvers = getMergedPolarisResolvers(
+        polarisSchemaConfig.addPolarisGraphQLScalars,
+        resolvers,
+    );
     schemaDirectives
```
Comment (chenshoo) [inline packages/polaris-schema/src/utils/executable-schema-creator.ts]: here should be the check not to add directives of permission if its disabled

File: packages/polaris-schema/src/utils/merge-types.ts [hunk 4]
```diff
@@ -2,17 +2,24 @@ import { ITypeDefinitions } from 'graphql-tools';
...
+    types: ITypeDefinitions,
+): string => {
+    const directivesAndScalarsTypeDefs = getDirectivesAndScalarsTypeDefs(polarisSchemaConfig);
+    return mergeTypes(
         [
+            ...directivesAndScalarsTypeDefs,
             repositoryEntityTypeDefs,
-            scalarsTypeDefs,
-            directivesTypeDefs,
             pageInfoTypeDef,
```
Comment (chenshoo) [inline packages/polaris-schema/src/utils/merge-types.ts]: check if this 3 is mandory, cause u didn't make a configuration for them
pageInfoTypeDef,
onlinePagingInputTypeDefs
filtersTypeDefs

File: packages/polaris-schema/src/scalars/polaris-scalars-resolvers.ts (hunk 1 above)
Comment (ronkatz96) [reply packages/polaris-schema/src/scalars/polaris-scalars-resolvers.ts:16]: only type def is needed. it is given by apollo server

**Comment by chenshoo** [comment]
add int tests

**Pull request merged**